src/rest/validateHandler.go

```diff
@@ -0,0 +1,107 @@
+package rest
+
+import (
+	"devicemanager/config"
+	"devicemanager/rest/redfish"
+	"encoding/json"
+	"fmt"
+	"github.com/kataras/iris/v12"
+	"github.com/kataras/iris/v12/context"
+	"github.com/sirupsen/logrus"
+	"net/http"
+)
+
+type validateHandler struct {
+	cfg config.Config
+}
+
+func (v *validateHandler) handle(ctx iris.Context) {
+	if ctx.GetHeader("Authorization") == "" {
+		ctx.StatusCode(http.StatusUnauthorized)
+		logrus.Error("No valid authorization")
+		ctx.WriteString("No valid authorization")
+		return
+	}
+
+	var reqInfo redfish.RequestInformation
+
+	// Retrieve request information from a request.
+	err := ctx.ReadJSON(&reqInfo)
+	if err != nil {
+		errorMessage := "Unable to retrieve mandatory information from a request: " + err.Error()
+		logrus.Error(errorMessage)
+		ctx.StatusCode(http.StatusBadRequest)
+		ctx.WriteString(errorMessage)
+		return
+	}
+
+	httpClient := redfish.NewHttpClient(v.cfg).WithBasicAuth(reqInfo.Username, string(reqInfo.Password))
+	serviceRootUri := fmt.Sprintf("https://%s%s", reqInfo.Host, "/redfish/v1")
+
+	response, err := httpClient.Get(serviceRootUri)
+	if err != nil {
+		errorMessage := "GET action failed due to: " + err.Error()
+		logrus.Error(errorMessage)
+		if response == nil {
+			ctx.StatusCode(http.StatusInternalServerError)
+			ctx.WriteString(errorMessage)
+			return
+		}
+	}
+
+	systemsUri := fmt.Sprintf("https://%s%s", reqInfo.Host, "/redfish/v1/Systems")
+
+	resp, err := httpClient.Get(systemsUri)
+	if err != nil {
+		errorMessage := "GET action failed due to: " + err.Error()
+		logrus.Error(errorMessage)
+		if response == nil {
+			ctx.StatusCode(http.StatusInternalServerError)
+			ctx.WriteString(errorMessage)
+			return
+		}
+	}
+
+	defer resp.Body.Close()
+	serviceRoot := &ServiceRoot{}
+	err = json.NewDecoder(resp.Body).Decode(serviceRoot)
+	if err != nil {
+		errorMessage := "Error while reading response body: " + err.Error()
+		logrus.Error(errorMessage)
+		ctx.StatusCode(http.StatusInternalServerError)
```

Comment:
this fragment of code is repeated 3 times in this func

Reply:
It is already fixed.